hs/src/Reach/APICut.hs

```diff
@@ -318,8 +317,8 @@ many_ = getFirst . mconcat . map First
 _many :: (a -> App (Maybe a)) -> [a] -> App (Maybe a)
 _many f l = many_ <$> mapM f l
 
-apc :: HasCounter a => a -> SLPart -> EPProg -> IO EPProg
-apc hc eWho = \case
+apc :: HasCounter a => a -> (SLPart, Maybe Int) -> EPProg -> IO EPProg
+apc hc (eWho, _) = \case
```

Comment:
I need to be convinced more about the changes to this program... how do you cut the EPProg at the right part? Is it that EPP has "pre-cut" it for you? If so, then I think you could still leave in those `err` cases, but you'd turn them into `impossible`

Reply:
Yes, EPP will essentially pre-cut the programs. The `EPProg` associated with `(who, step)` will have removed every other API call that is not in `step`. 

I don't think the err cases should be impossible. For example, if there are two paths to calling `interact.out` (from a conditional), `eSeenOut` can be true

Reply:
First thing, that's cool.

Second thing, you're taking something that used to be an error path and making it okay. The error was there because something bad would happen. The pre-cut argument suggests that the bad thing I'm worried about is impossible. Thus, I suggest impossible. Now, you suggest that maybe we've enabled some new behavior that would not be bad. If that's the case, I want a test case that demonstrates that new feature. My taste is to turn it into an impossible, come up with a test case that triggers it, decide if we want to allow the program, remove the impossible, and ensure that the program does what we think it should

Reply:
okay Ill look into that